financial-connections/src/main/java/com/stripe/android/financialconnections/features/error/ErrorScreen.kt

```diff
@@ -0,0 +1,116 @@
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane
+import com.stripe.android.financialconnections.presentation.parentViewModel
+import com.stripe.android.financialconnections.ui.FinancialConnectionsPreview
+import com.stripe.android.financialconnections.ui.components.FinancialConnectionsScaffold
+import com.stripe.android.financialconnections.ui.components.FinancialConnectionsTopAppBar
+
+@Composable
+internal fun ErrorScreen() {
+    val viewModel: ErrorViewModel = mavericksViewModel()
+    val parentViewModel = parentViewModel()
+    val payload = viewModel.collectAsState { it.payload }
+    BackHandler(enabled = true) {}
+    ErrorContent(
+        payload = payload.value,
+        onManualEntryClick = viewModel::onManualEntryClick,
+        onSelectBankClick = viewModel::onSelectBankClick,
+        onCloseClick = { parentViewModel.onCloseWithConfirmationClick(Pane.UNEXPECTED_ERROR) },
+        onCloseFromErrorClick = parentViewModel::onCloseFromErrorClick
+    )
+}
+
+@Composable
+private fun ErrorContent(
+    payload: Async<ErrorState.Payload>,
+    onSelectBankClick: () -> Unit,
+    onManualEntryClick: () -> Unit,
+    onCloseClick: () -> Unit,
+    onCloseFromErrorClick: (Throwable) -> Unit
+) {
+    FinancialConnectionsScaffold(
+        topBar = {
+            FinancialConnectionsTopAppBar(
+                showBack = false,
+                onCloseClick = onCloseClick
+            )
+        }
+    ) {
+        when (payload) {
+            Uninitialized,
+            is Loading -> FullScreenGenericLoading()
+            is Success -> ErrorContent(
+                payload().error,
+                onSelectAnotherBank = onSelectBankClick,
+                onEnterDetailsManually = onManualEntryClick,
+                onCloseFromErrorClick = onCloseFromErrorClick
+            )
+            is Fail -> UnclassifiedErrorContent(
+                error = payload.error,
+                onCloseFromErrorClick = onCloseFromErrorClick
+            )
+        }
+    }
+}
+
+@Composable
+private fun ErrorContent(
+    error: Throwable,
+    onSelectAnotherBank: () -> Unit,
+    onEnterDetailsManually: () -> Unit,
+    onCloseFromErrorClick: (Throwable) -> Unit
+) {
+    when (error) {
+        is InstitutionPlannedDowntimeError -> InstitutionPlannedDowntimeErrorContent(
+            exception = error,
+            onSelectAnotherBank = onSelectAnotherBank,
+            onEnterDetailsManually = onEnterDetailsManually
+        )
+
+        is InstitutionUnplannedDowntimeError -> InstitutionUnplannedDowntimeErrorContent(
+            exception = error,
+            onSelectAnotherBank = onSelectAnotherBank,
+            onEnterDetailsManually = onEnterDetailsManually
+        )
+
+        is PartnerAuthError -> InstitutionUnknownErrorContent(
+            onSelectAnotherBank = onSelectAnotherBank,
+        )
+        else -> UnclassifiedErrorContent(error, onCloseFromErrorClick)
+    }
+}
```

Comment:
This should extend to handle all specific errors accordingly (this is what web does)